com.unity.shadergraph/Editor/Generation/Processors/ActiveFields.cs

```diff
@@ -19,8 +20,49 @@ internal interface IActiveFieldsSet : KeywordDependentCollection.ISet<IActiveFie
         void AddAll(FieldDescriptor field);
     }
 
+    internal class FieldNamePairStorage
+    {
+        private HashSet<FieldDescriptor> m_fieldDescriptors;
+        private HashSet<string> m_fieldNames;
+
+        public IEnumerable<FieldDescriptor> fields => m_fieldDescriptors;
+
+        public FieldNamePairStorage()
+        {
+            m_fieldDescriptors = new HashSet<FieldDescriptor>();
+            m_fieldNames = new HashSet<string>(StringComparer.Ordinal);
+        }
+
+        public IEnumerable<FieldDescriptor> Union(FieldNamePairStorage other)
+        {
+            var output =  new HashSet<FieldDescriptor>(m_fieldDescriptors);
+            output.UnionWith(other.m_fieldDescriptors);
+            return output;
+        }
+
+        public bool Contains(FieldDescriptor fieldDescriptor)
```

Comment:
Does anyone ever look up by FieldDescriptor?

Reply:
Shadergraph does in a number of places yes